samples/bootloader/src/main.c

```diff
@@ -92,8 +92,12 @@ void main(void)
 	uint32_t s1_addr = s1_address_read();
 	const struct fw_info *s0_info = fw_info_find(s0_addr);
 	const struct fw_info *s1_info = fw_info_find(s1_addr);
+	slot_priority priority = slot_priority_from_image_trailers();
 
-	if (!s1_info || (s0_info->version >= s1_info->version)) {
+	if (!s1_info && (s0_info->version > s1_info->version)) {
```

Comment:
You're gonna access a bad pointer here.
I think, if we really want to preserve behavior here, we might actually need an "undecided" priority to check against.
```
if (priority == NONE) {
    if (!s1_info || (s0_info->version >= s1_info->version))
```

Alternatively actually you might just need to not flip the `||` to `&&` and keep the remaining logic where you substitute `>=` for `>`
This does mean that we'll _always_ prioritize the bigger version of S0 though. So then you probably need an equivalent prioritization for if the version of S1 is greater than S0.

Maybe
```
if (s1_info && (s0_info->version != s1_info->version)) {
    if (s0_info->version > s1_info->version)
        priority = SLOT_PRIORITY_S0;
    else if (s1_info->version > s0_info->version)
        priority = SLOT_PRIORITY_S1;
}
```